hyperspy/io_plugins/digital_micrograph.py

```diff
@@ -807,10 +807,14 @@ def _get_microscope_name(self, ImageTags):
             if ImageTags.Session_Info.Microscope != "[]":
                 return ImageTags.Session_Info.Microscope
         except AttributeError:
-            if 'Name' in ImageTags['Microscope_Info'].keys():
-                return ImageTags.Microscope_Info.Name
-            elif 'Microscope' in ImageTags['Microscope_Info'].keys():
-                return ImageTags.Microscope_Info.Microscope
+            try:
+                if 'Name' in ImageTags['Microscope_Info'].keys():
+                    return ImageTags.Microscope_Info.Name
+                elif 'Microscope' in ImageTags['Microscope_Info'].keys():
+                    return ImageTags.Microscope_Info.Microscope
+            except:
+                _logger.info("Microscope name not present")
+                return None
```

Comment:
Is it worth specifying?
```python 
return None
```
Isn't python doing it by default?

Reply:
@ericpre is correct.

Reply:
What's about changing this to:
```python
def _get_microscope_name(self, ImageTags):
    # This function is called only when the 'Microscope_Info' tag is present
    if "Session_Info" in ImageTags.keys() and len(ImageTags.Session_Info.Microscope) > 0:
        return ImageTags.Session_Info.Microscope
    if 'Name' in ImageTags['Microscope_Info'].keys():
        return ImageTags.Microscope_Info.Name
    if 'Microscope' in ImageTags['Microscope_Info'].keys():
        return ImageTags.Microscope_Info.Microscope
    # in case none of them above works add this message to the logger.
    _logger.info("Microscope name not present.")
```
I find it easier to read.

Reply:
I would suggest the following (non-equivalent code) instead:

```python
def _get_microscope_name(self, ImageTags):
    # This function is called only when the 'Microscope_Info' tag is present
    if "Session_Info" in ImageTags.keys() andImageTags.Session_Info.Microscope:
        return ImageTags.Session_Info.Microscope
    elif 'Name' in ImageTags['Microscope_Info'].keys() and ImageTags.Microscope_Info.Name:
        return ImageTags.Microscope_Info.Name
    elif 'Microscope' in ImageTags['Microscope_Info'].keys() and ImageTags.Microscope_Info.Microscope:
        return ImageTags.Microscope_Info.Microscope
    else:
        # in case none of them above works add this message to the logger.
        _logger.info("Microscope name not present.")
        return None
```

``return None`` is unnecessary, but it is easier to understand for somebody who doesn't know that Python does it for you, so I would leave it to the author's taste.

Reply:
Yes, I agree, that writing `return None` make the code more clear.
Regarding the `if "Session_Info" in ImageTags.keys() and ImageTags.Session_Info.Microscope:`, it can exist and be empty, which is the case by default. AFAIK it needs to be set by the user in the global info of DM. However, in case, this is an empty list or string it is worth reading the microscope name from the 'Microscope_Info' which has a default value when connected to a microscope (for example, something like "FEI Tecnai" or "FEI Tecnai Remote" on any FEI microscope including Titan and other.)
Actually, I am wondering if this is worth parsing `ImageTags.Microscope_Info.Microscope` since most of the time it will set to a wrong default value...